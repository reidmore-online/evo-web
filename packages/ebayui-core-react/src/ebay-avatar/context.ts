import { createContext } from "react";

export const AvatarContext = createContext<{
    setImagePlacement: (value: "fit" | "cover") => void;
} | null>(null);
