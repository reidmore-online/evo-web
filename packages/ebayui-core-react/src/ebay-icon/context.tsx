import React, { createContext, type ReactNode } from "react";

export const IconContext = createContext<Set<string>>(null);

export const ROOT_ID = "ebayui-svg-symbols";
export const EbayIconProvider = ({ children }: { children: ReactNode }) => (
    <IconContext.Provider value={new Set()}>
        <svg
            id={ROOT_ID}
            style={{ position: "absolute", height: "0px", width: "0px" }}
            focusable={false}
            aria-hidden="true"
        />
        {children}
    </IconContext.Provider>
);
