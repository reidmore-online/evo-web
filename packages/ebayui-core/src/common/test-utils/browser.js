import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el, to, cb) {
    fireEvent.scroll(el);
    el.scrollLeft = to;
    setTimeout(cb, 600);
}

function waitFrames(count, cb) {
    if (count) {
        return requestAnimationFrame(() => {
            waitFrames(count - 1, cb);
        });
    }

    cb();
}

async function pressKey(el, info) {
    for (const event of [
        createEvent.keyDown(el, info),
        createEvent.keyUp(el, info),
    ]) {
        // we assign properties to them for older browsers (chrome 49)
        Object.keys(info).forEach((key) => {
            if (event[key] !== info[key]) {
                Object.defineProperty(event, key, { value: info[key] });
            }
        });

        await fireEvent(el, event);
    }
}

const fastAnimations = {
    // Adds an style to the document which forces all transitions to run more quickly for the tests.
    start(disable) {
        if (this.fastAnimationStyle) {
            return;
        }
        const time = disable ? 0 : 1;

        this.fastAnimationStyle = document.createElement("style");
        this.fastAnimationStyle.innerHTML = `*,details::details-content,*::before,*::after {
                animation-duration: ${time}ms !important;
                transition-duration: ${time}ms !important;
                transition-delay: ${time}ms !important;
                scroll-behavior: auto !important;
            }`;
        document.head.appendChild(this.fastAnimationStyle);
    },
    stop() {
        if (!this.fastAnimationStyle) {
            return;
        }

        document.head.removeChild(this.fastAnimationStyle);
        this.fastAnimationStyle = undefined;
    },
};

export { fastAnimations, waitFrames, pressKey, simulateScroll };
