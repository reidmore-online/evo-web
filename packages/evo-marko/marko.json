{
  "taglib-id": "evo-marko",
  "tags-dir": "./src/tags",
  "taglib-imports": [
    "./src/tags/evo-icon/marko.json",
    "./src/tags/evo-skeleton/marko.json"
  ]
}
