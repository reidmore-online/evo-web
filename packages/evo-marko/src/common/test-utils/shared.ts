export function getNItems(n: number, getAttrs: (i: number) => any[]) {
  return Array.from({ length: n }).map((_, i) => getAttrs(i));
}

// Helper method to render pagination items
export function getPaginationItems(
  n: number,
  href: boolean = false,
  selected: number = -1,
  navDisabled: boolean = false,
  skipPrevNext: boolean = false,
) {
  const items = [];
  if (!skipPrevNext) {
    items.push({
      type: "previous",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  for (let i = 0; i < n; i++) {
    items.push({
      href: href ? "#" : null,
      current: i === selected,
    });
  }
  if (!skipPrevNext) {
    items.push({
      type: "next",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  return items;
}
