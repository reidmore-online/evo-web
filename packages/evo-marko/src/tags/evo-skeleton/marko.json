{
  "taglib-id": "evo-marko-skeleton",
  "tags-dir": "./tags"
}
