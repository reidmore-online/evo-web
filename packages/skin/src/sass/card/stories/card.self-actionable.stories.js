export default { title: "Skin/Card/Self Actionable" };

export const verticalButton = () => `
<button class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const verticalLink = () => `
<a class="card" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;

export const horizontalButton = () => `
<button class="card card--horizontal">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const horizontalLink = () => `
<a class="card card--horizontal" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;
