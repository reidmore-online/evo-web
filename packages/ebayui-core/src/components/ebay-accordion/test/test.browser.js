import {
    afterEach,
    beforeEach,
    afterAll,
    beforeAll,
    describe,
    it,
    expect,
} from "vitest";
import { render, cleanup } from "@marko/testing-library";
import { userEvent } from "@vitest/browser/context";
import { fastAnimations } from "../../../common/test-utils/browser";
import { diffHTML, visualHTML } from "../../../common/test-utils/snapshots";
import { composeStories } from "@storybook/marko";
import * as stories from "../accordion.stories"; // import all stories from the stories file
const { Default, AutoCollapsed } = composeStories(stories);

beforeAll(() => fastAnimations.start(true));
afterAll(() => fastAnimations.stop());
afterEach(cleanup);
let component;
let user;

describe("accordion", () => {
    beforeEach(() => {
        user = userEvent.setup();
    });
    afterEach(() => {
        user.cleanup();
    });
    describe("given the accordion in the default state", () => {
        beforeEach(async () => {
            component = await render(Default);
        });

        it("then it matches the snapshot", () => {
            const html = visualHTML(component.container);

            expect(html).toMatchSnapshot();
        });

        describe("when first section toggled", () => {
            let initialHTML;
            beforeEach(async () => {
                initialHTML = diffHTML(component.container);
                await user.click(component.getByText("Item 1"));
            });

            it("should open the clicked section", async () => {
                expect(initialHTML(component.container)).toMatchSnapshot();
            });

            describe("when another section is opened", () => {
                beforeEach(async () => {
                    await user.click(component.getByText("Item 1"));
                    await user.hover(component.getByText("Item 1"));
                });

                it("should close an open section when clicked again", async () => {
                    expect(initialHTML(component.container)).toMatchSnapshot(
                        "Should have no changes from initial",
                    );
                });
            });
        });
    });

    describe("given the accordion with autocollapse enabled", () => {
        let initialHTML;
        beforeEach(async () => {
            component = await render(AutoCollapsed);
            initialHTML = diffHTML(component.container);
        });
        describe("when first section toggled", () => {
            beforeEach(async () => {
                await user.click(component.getByText("Item 1"));
            });

            it("should open as normal", async () => {
                expect(initialHTML(component.container)).toMatchSnapshot();
            });

            describe("when another section is opened", () => {
                beforeEach(async () => {
                    initialHTML = diffHTML(component.container);
                    // Open second section
                    await user.click(component.getByText("Item 2"));
                });
                it("should collapse previous section when new section is opened", async () => {
                    // Verify first section closed and second section opened
                    expect(initialHTML(component.container)).toMatchSnapshot();
                });
            });
        });
    });
});
