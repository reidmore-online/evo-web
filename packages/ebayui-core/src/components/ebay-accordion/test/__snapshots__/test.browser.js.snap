// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`accordion > given the accordion in the default state > then it matches the snapshot 1`] = `
"<div style="
  animation-duration: 0ms;
  transition-delay: 0ms;
  transition-duration: 0ms
">
  <ul style="
    animation-duration: 0ms;
    margin-bottom: 0px;
    margin-top: 0px;
    padding-left: 0px;
    transition-delay: 0ms;
    transition-duration: 0ms;
    width: 100%
  ">
    <li style="
      animation-duration: 0ms;
      border-bottom: 1px solid var(--color-stroke-subtle);
      transition-delay: 0ms;
      transition-duration: 0ms
    ">
      <style>@scope{:scope{
        &::marker {font-size: 0px}
      }}</style>
      <details style="
        animation-duration: 0ms;
        overflow-x: hidden;
        overflow-y: hidden;
        transition-delay: 0ms;
        transition-duration: 0ms
      ">
        <style>@scope{:scope{
          &::details-content {
            animation-duration: 0ms;
            opacity: 0;
            overflow-x: clip;
            overflow-y: clip;
            transition-delay: 0ms;
            transition-duration: 0ms;
            transition: height var(--motion-duration-medium-3) var(--motion-easing-standard),
                        content-visibility var(--motion-duration-medium-2)
                            var(--motion-easing-continuous) allow-discrete,
                        opacity var(--motion-duration-medium-2)
                            var(--motion-easing-continuous)
          }
          &::marker {font-size: 0px}
        }}</style>
        <summary style="
          align-items: center;
          animation-duration: 0ms;
          box-sizing: border-box;
          color: var(
                  --details-summary-foreground-color,
                  var(--color-foreground-primary)
              );
          display: flex;
          font-size: var(--font-size-medium);
          font-weight: var(--font-weight-bold);
          justify-content: space-between;
          list-style-position: inside;
          list-style-type: none;
          min-height: 48px;
          padding-bottom: 12px;
          padding-left: 16px;
          padding-right: 16px;
          padding-top: 12px;
          transition-delay: 0ms;
          transition-duration: 0ms
        ">
          <style>@scope{:scope{
            &::marker {font-size: 0px}
          }}</style>
          <span style="
            animation-duration: 0ms;
            transition-delay: 0ms;
            transition-duration: 0ms
          ">
            <style>@scope{:scope{
              &::marker {font-size: 0px}
            }}</style>
            Item 1
          </span>
          <span
            hidden
            style="
              animation-duration: 0ms;
              display: inline-flex;
              margin-inline-start: 8px;
              transition-delay: 0ms;
              transition-duration: 0ms
            "
          >
            <style>@scope{:scope{
              &::marker {font-size: 0px}
            }}</style>
            <svg
              focusable="false"
              style="
                animation-duration: 0ms;
                animation: chevron-bounce-close var(--motion-duration-medium-1)
                        var(--motion-easing-standard);
                display: inline-block;
                fill: var(
                        --details-secondary-foreground-color,
                        var(--color-foreground-secondary)
                    );
                height: 16px;
                pointer-events: none;
                stroke-width: 0;
                stroke: currentcolor;
                transition-delay: 0ms;
                transition-duration: 0ms;
                vertical-align: middle;
                width: 16px
              "
            >
              <style>@scope{:scope{
                &::marker {font-size: 0px}
              }}</style>
              <use
                href="#icon-chevron-down-16"
                style="
                  animation-duration: 0ms;
                  transition-delay: 0ms;
                  transition-duration: 0ms
                "
              >
                <style>@scope{:scope{
                  &::marker {font-size: 0px}
                }}</style>
              </use>
            </svg>
          </span>
        </summary>
        <div style="
          animation-duration: 0ms;
          margin-bottom: 6px;
          margin-left: 16px;
          margin-right: 16px;
          overflow-x: hidden;
          overflow-y: hidden;
          transition-delay: 0ms;
          transition-duration: 0ms
        ">
          <style>@scope{:scope{
            &::marker {font-size: 0px}
          }}</style>
          <p style="
            animation-duration: 0ms;
            transition-delay: 0ms;
            transition-duration: 0ms
          ">
            <style>@scope{:scope{
              &::marker {font-size: 0px}
            }}</style>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li style="
      animation-duration: 0ms;
      border-bottom: 1px solid var(--color-stroke-subtle);
      transition-delay: 0ms;
      transition-duration: 0ms
    ">
      <style>@scope{:scope{
        &::marker {font-size: 0px}
      }}</style>
      <details style="
        animation-duration: 0ms;
        overflow-x: hidden;
        overflow-y: hidden;
        transition-delay: 0ms;
        transition-duration: 0ms
      ">
        <style>@scope{:scope{
          &::details-content {
            animation-duration: 0ms;
            opacity: 0;
            overflow-x: clip;
            overflow-y: clip;
            transition-delay: 0ms;
            transition-duration: 0ms;
            transition: height var(--motion-duration-medium-3) var(--motion-easing-standard),
                        content-visibility var(--motion-duration-medium-2)
                            var(--motion-easing-continuous) allow-discrete,
                        opacity var(--motion-duration-medium-2)
                            var(--motion-easing-continuous)
          }
          &::marker {font-size: 0px}
        }}</style>
        <summary style="
          align-items: center;
          animation-duration: 0ms;
          box-sizing: border-box;
          color: var(
                  --details-summary-foreground-color,
                  var(--color-foreground-primary)
              );
          display: flex;
          font-size: var(--font-size-medium);
          font-weight: var(--font-weight-bold);
          justify-content: space-between;
          list-style-position: inside;
          list-style-type: none;
          min-height: 48px;
          padding-bottom: 12px;
          padding-left: 16px;
          padding-right: 16px;
          padding-top: 12px;
          transition-delay: 0ms;
          transition-duration: 0ms
        ">
          <style>@scope{:scope{
            &::marker {font-size: 0px}
          }}</style>
          <span style="
            animation-duration: 0ms;
            transition-delay: 0ms;
            transition-duration: 0ms
          ">
            <style>@scope{:scope{
              &::marker {font-size: 0px}
            }}</style>
            Item 2
          </span>
          <span
            hidden
            style="
              animation-duration: 0ms;
              display: inline-flex;
              margin-inline-start: 8px;
              transition-delay: 0ms;
              transition-duration: 0ms
            "
          >
            <style>@scope{:scope{
              &::marker {font-size: 0px}
            }}</style>
            <svg
              focusable="false"
              style="
                animation-duration: 0ms;
                animation: chevron-bounce-close var(--motion-duration-medium-1)
                        var(--motion-easing-standard);
                display: inline-block;
                fill: var(
                        --details-secondary-foreground-color,
                        var(--color-foreground-secondary)
                    );
                height: 16px;
                pointer-events: none;
                stroke-width: 0;
                stroke: currentcolor;
                transition-delay: 0ms;
                transition-duration: 0ms;
                vertical-align: middle;
                width: 16px
              "
            >
              <style>@scope{:scope{
                &::marker {font-size: 0px}
              }}</style>
              <use
                href="#icon-chevron-down-16"
                style="
                  animation-duration: 0ms;
                  transition-delay: 0ms;
                  transition-duration: 0ms
                "
              >
                <style>@scope{:scope{
                  &::marker {font-size: 0px}
                }}</style>
              </use>
            </svg>
          </span>
        </summary>
        <div style="
          animation-duration: 0ms;
          margin-bottom: 6px;
          margin-left: 16px;
          margin-right: 16px;
          overflow-x: hidden;
          overflow-y: hidden;
          transition-delay: 0ms;
          transition-duration: 0ms
        ">
          <style>@scope{:scope{
            &::marker {font-size: 0px}
          }}</style>
          <p style="
            animation-duration: 0ms;
            transition-delay: 0ms;
            transition-duration: 0ms
          ">
            <style>@scope{:scope{
              &::marker {font-size: 0px}
            }}</style>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li style="
      animation-duration: 0ms;
      transition-delay: 0ms;
      transition-duration: 0ms
    ">
      <style>@scope{:scope{
        &::marker {font-size: 0px}
      }}</style>
      <details style="
        animation-duration: 0ms;
        overflow-x: hidden;
        overflow-y: hidden;
        transition-delay: 0ms;
        transition-duration: 0ms
      ">
        <style>@scope{:scope{
          &::details-content {
            animation-duration: 0ms;
            opacity: 0;
            overflow-x: clip;
            overflow-y: clip;
            transition-delay: 0ms;
            transition-duration: 0ms;
            transition: height var(--motion-duration-medium-3) var(--motion-easing-standard),
                        content-visibility var(--motion-duration-medium-2)
                            var(--motion-easing-continuous) allow-discrete,
                        opacity var(--motion-duration-medium-2)
                            var(--motion-easing-continuous)
          }
          &::marker {font-size: 0px}
        }}</style>
        <summary style="
          align-items: center;
          animation-duration: 0ms;
          box-sizing: border-box;
          color: var(
                  --details-summary-foreground-color,
                  var(--color-foreground-primary)
              );
          display: flex;
          font-size: var(--font-size-medium);
          font-weight: var(--font-weight-bold);
          justify-content: space-between;
          list-style-position: inside;
          list-style-type: none;
          min-height: 48px;
          padding-bottom: 12px;
          padding-left: 16px;
          padding-right: 16px;
          padding-top: 12px;
          transition-delay: 0ms;
          transition-duration: 0ms
        ">
          <style>@scope{:scope{
            &::marker {font-size: 0px}
          }}</style>
          <span style="
            animation-duration: 0ms;
            transition-delay: 0ms;
            transition-duration: 0ms
          ">
            <style>@scope{:scope{
              &::marker {font-size: 0px}
            }}</style>
            Item 3
          </span>
          <span
            hidden
            style="
              animation-duration: 0ms;
              display: inline-flex;
              margin-inline-start: 8px;
              transition-delay: 0ms;
              transition-duration: 0ms
            "
          >
            <style>@scope{:scope{
              &::marker {font-size: 0px}
            }}</style>
            <svg
              focusable="false"
              style="
                animation-duration: 0ms;
                animation: chevron-bounce-close var(--motion-duration-medium-1)
                        var(--motion-easing-standard);
                display: inline-block;
                fill: var(
                        --details-secondary-foreground-color,
                        var(--color-foreground-secondary)
                    );
                height: 16px;
                pointer-events: none;
                stroke-width: 0;
                stroke: currentcolor;
                transition-delay: 0ms;
                transition-duration: 0ms;
                vertical-align: middle;
                width: 16px
              "
            >
              <style>@scope{:scope{
                &::marker {font-size: 0px}
              }}</style>
              <use
                href="#icon-chevron-down-16"
                style="
                  animation-duration: 0ms;
                  transition-delay: 0ms;
                  transition-duration: 0ms
                "
              >
                <style>@scope{:scope{
                  &::marker {font-size: 0px}
                }}</style>
              </use>
            </svg>
          </span>
        </summary>
        <div style="
          animation-duration: 0ms;
          margin-bottom: 6px;
          margin-left: 16px;
          margin-right: 16px;
          overflow-x: hidden;
          overflow-y: hidden;
          transition-delay: 0ms;
          transition-duration: 0ms
        ">
          <style>@scope{:scope{
            &::marker {font-size: 0px}
          }}</style>
          <p style="
            animation-duration: 0ms;
            transition-delay: 0ms;
            transition-duration: 0ms
          ">
            <style>@scope{:scope{
              &::marker {font-size: 0px}
            }}</style>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
  </ul>
</div>"
`;

exports[`accordion > given the accordion in the default state > when first section toggled > should open the clicked section 1`] = `
"@@ -21,17 +21,20 @@
       <style>@scope{:scope{
         &::marker {font-size: 0px}
       }}</style>
-       <details style="
-         animation-duration: 0ms;
-         overflow-x: hidden;
-         overflow-y: hidden;
-         transition-delay: 0ms;
-         transition-duration: 0ms
-       ">
+       <details
+         open
+         style="
+           animation-duration: 0ms;
+           overflow-x: hidden;
+           overflow-y: hidden;
+           transition-delay: 0ms;
+           transition-duration: 0ms
+         "
+       >
         <style>@scope{:scope{
           &::details-content {
             animation-duration: 0ms;
-             opacity: 0;
+             height: auto;
             overflow-x: clip;
             overflow-y: clip;
             transition-delay: 0ms;
@@ -47,6 +50,7 @@
         <summary style="
           align-items: center;
           animation-duration: 0ms;
+           background-color: var(--color-state-secondary-hover);
           box-sizing: border-box;
           color: var(
                   --details-summary-foreground-color,
@@ -85,6 +89,7 @@
               animation-duration: 0ms;
               display: inline-flex;
               margin-inline-start: 8px;
+               transform: rotate(180deg);
               transition-delay: 0ms;
               transition-duration: 0ms
             "
@@ -96,8 +101,8 @@
               focusable="false"
               style="
                 animation-duration: 0ms;
-                 animation: chevron-bounce-close var(--motion-duration-medium-1)
-                         var(--motion-easing-standard);
+                 animation: chevron-bounce-open var(--motion-duration-medium-1)
+                         var(--motion-easing-standard) var(--motion-duration-short-1);
                 display: inline-block;
                 fill: var(
                         --details-secondary-foreground-color,
"
`;

exports[`accordion > given the accordion in the default state > when first section toggled > when another section is opened > should close an open section when clicked again > Should have no changes from initial 1`] = `
"@@ -47,6 +47,7 @@
         <summary style="
           align-items: center;
           animation-duration: 0ms;
+           background-color: var(--color-state-secondary-hover);
           box-sizing: border-box;
           color: var(
                   --details-summary-foreground-color,
"
`;

exports[`accordion > given the accordion with autocollapse enabled > when first section toggled > should open as normal 1`] = `
"@@ -21,17 +21,20 @@
       <style>@scope{:scope{
         &::marker {font-size: 0px}
       }}</style>
-       <details style="
-         animation-duration: 0ms;
-         overflow-x: hidden;
-         overflow-y: hidden;
-         transition-delay: 0ms;
-         transition-duration: 0ms
-       ">
+       <details
+         open
+         style="
+           animation-duration: 0ms;
+           overflow-x: hidden;
+           overflow-y: hidden;
+           transition-delay: 0ms;
+           transition-duration: 0ms
+         "
+       >
         <style>@scope{:scope{
           &::details-content {
             animation-duration: 0ms;
-             opacity: 0;
+             height: auto;
             overflow-x: clip;
             overflow-y: clip;
             transition-delay: 0ms;
@@ -47,6 +50,7 @@
         <summary style="
           align-items: center;
           animation-duration: 0ms;
+           background-color: var(--color-state-secondary-hover);
           box-sizing: border-box;
           color: var(
                   --details-summary-foreground-color,
@@ -85,6 +89,7 @@
               animation-duration: 0ms;
               display: inline-flex;
               margin-inline-start: 8px;
+               transform: rotate(180deg);
               transition-delay: 0ms;
               transition-duration: 0ms
             "
@@ -96,8 +101,8 @@
               focusable="false"
               style="
                 animation-duration: 0ms;
-                 animation: chevron-bounce-close var(--motion-duration-medium-1)
-                         var(--motion-easing-standard);
+                 animation: chevron-bounce-open var(--motion-duration-medium-1)
+                         var(--motion-easing-standard) var(--motion-duration-short-1);
                 display: inline-block;
                 fill: var(
                         --details-secondary-foreground-color,
"
`;

exports[`accordion > given the accordion with autocollapse enabled > when first section toggled > when another section is opened > should collapse previous section when new section is opened 1`] = `
"@@ -21,20 +21,17 @@
       <style>@scope{:scope{
         &::marker {font-size: 0px}
       }}</style>
-       <details
-         open
-         style="
-           animation-duration: 0ms;
-           overflow-x: hidden;
-           overflow-y: hidden;
-           transition-delay: 0ms;
-           transition-duration: 0ms
-         "
-       >
+       <details style="
+         animation-duration: 0ms;
+         overflow-x: hidden;
+         overflow-y: hidden;
+         transition-delay: 0ms;
+         transition-duration: 0ms
+       ">
         <style>@scope{:scope{
           &::details-content {
             animation-duration: 0ms;
-             height: auto;
+             opacity: 0;
             overflow-x: clip;
             overflow-y: clip;
             transition-delay: 0ms;
@@ -50,7 +47,6 @@
         <summary style="
           align-items: center;
           animation-duration: 0ms;
-           background-color: var(--color-state-secondary-hover);
           box-sizing: border-box;
           color: var(
                   --details-summary-foreground-color,
@@ -89,7 +85,6 @@
               animation-duration: 0ms;
               display: inline-flex;
               margin-inline-start: 8px;
-               transform: rotate(180deg);
               transition-delay: 0ms;
               transition-duration: 0ms
             "
@@ -101,8 +96,8 @@
               focusable="false"
               style="
                 animation-duration: 0ms;
-                 animation: chevron-bounce-open var(--motion-duration-medium-1)
-                         var(--motion-easing-standard) var(--motion-duration-short-1);
+                 animation: chevron-bounce-close var(--motion-duration-medium-1)
+                         var(--motion-easing-standard);
                 display: inline-block;
                 fill: var(
                         --details-secondary-foreground-color,
@@ -171,17 +166,20 @@
       <style>@scope{:scope{
         &::marker {font-size: 0px}
       }}</style>
-       <details style="
-         animation-duration: 0ms;
-         overflow-x: hidden;
-         overflow-y: hidden;
-         transition-delay: 0ms;
-         transition-duration: 0ms
-       ">
+       <details
+         open
+         style="
+           animation-duration: 0ms;
+           overflow-x: hidden;
+           overflow-y: hidden;
+           transition-delay: 0ms;
+           transition-duration: 0ms
+         "
+       >
         <style>@scope{:scope{
           &::details-content {
             animation-duration: 0ms;
-             opacity: 0;
+             height: auto;
             overflow-x: clip;
             overflow-y: clip;
             transition-delay: 0ms;
@@ -197,6 +195,7 @@
         <summary style="
           align-items: center;
           animation-duration: 0ms;
+           background-color: var(--color-state-secondary-hover);
           box-sizing: border-box;
           color: var(
                   --details-summary-foreground-color,
@@ -235,6 +234,7 @@
               animation-duration: 0ms;
               display: inline-flex;
               margin-inline-start: 8px;
+               transform: rotate(180deg);
               transition-delay: 0ms;
               transition-duration: 0ms
             "
@@ -246,8 +246,8 @@
               focusable="false"
               style="
                 animation-duration: 0ms;
-                 animation: chevron-bounce-close var(--motion-duration-medium-1)
-                         var(--motion-easing-standard);
+                 animation: chevron-bounce-open var(--motion-duration-medium-1)
+                         var(--motion-easing-standard) var(--motion-duration-short-1);
                 display: inline-block;
                 fill: var(
                         --details-secondary-foreground-color,
"
`;
